src/main/java/com/cloudbees/jenkins/support/impl/ThreadDumps.java

```diff
@@ -45,11 +48,41 @@
 @Extension(ordinal = -100.0) // run this last as it blocks the channel
 public class ThreadDumps extends ObjectComponent<Computer> {
 
-    private final Logger logger = Logger.getLogger(ThreadDumps.class.getName());
+    private static final String DATE_FORMAT = "yyyyMMdd-HHmmss.SSS";
+
+    private static final Logger LOGGER = Logger.getLogger(ThreadDumps.class.getName());
 
     @DataBoundConstructor
     public ThreadDumps() {}
 
+    /**
+     * Collects multiple thread dumps and saves them to the specified file list.
+     * @param fileList the file list where the thread dumps will be saved
+     * @param timestamp the initial timestamp
+     * @param delay the delay between two consecutive thread dumps
+     * @param iterations the number of thread dumps to collect
+     */
+    public static void collectMultiple(FileListCap fileList, long timestamp, long delay, int iterations) {
+        if (delay <= 0) {
+            throw new IllegalArgumentException("delay must be greater than 0");
+        }
+        var format = new SimpleDateFormat(DATE_FORMAT);
+        for (int i = 0; i < iterations; i++) {
+            File threadDumpFile = fileList.file(format.format(new Date(timestamp)) + ".txt");
```

Comment:
Suppressed the prefix `-i`as the timestamp is always different anyway

Reply:
(referring to code moved from `SlowRequestThreadDumpsGenerator`)